### PR TITLE
allow ttl for token

### PR DESCRIPTION
**Pull request by rajatjindal**
Allow TTL for the tokens

### CODE REVIEW

File: auth/token_issuer.go [hunk 1]
```diff
@@ -16,6 +18,7 @@ type LDAPTokenIssuer struct {
 	LDAPServer        string
 	LDAPAuthenticator ldap.Authenticator
 	TokenSigner       token.Signer
+	Ttl               time.Duration
```
Comment (anandsomani) [inline auth/token_issuer.go]: seems like the current code style uses upper case for abbr. so TTL

**Review by raghu67 — APPROVED** [review]
Looks good

**Comment by johngmyers** [comment]
I don't see any added tests for this functionality.

**Comment by rajatjindal** [comment]
@johngmyers added the testcases. can you please approve if looks ok

File: cmd/serve.go [hunk 2]
```diff
@@ -149,7 +149,7 @@ func validate() {
 	ldapSkipTlsVerification = viper.GetBool("ldap-skip-tls-verification")
 
 	tokenTtl = viper.GetDuration("token-ttl")
-
+
```
Comment (johngmyers) [inline cmd/serve.go]: nit: unnecessary change

File: auth/webhook.go [hunk 3]
```diff
@@ -42,6 +42,8 @@ func (tw *TokenWebhook) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
 	if err != nil {
 		glog.Errorf("Token is invalid: %v", err)
 		resp.WriteHeader(http.StatusUnauthorized)
+		resp.Header().Add("Content-Type", "text/plain")
+		resp.Write([]byte(err.Error()))
```
Comment (johngmyers) [inline auth/webhook.go:46]: Missing test for this.

File: start.sh [hunk 4]
```diff
@@ -1,30 +0,0 @@
...
-CMD = "$CMD --ldap-search-user-password $LDAP_SEARCH_USER_PASSWORD"
-CMD = "$CMD --tls-cert-file $TLS_CERT_FILE"
-CMD = "$CMD --tls-private-key $TLS_PRIVATE_KEY"
-CMD = "$CMD --ldap-user-attribute $LDAP_USER_ATTRIBUTE"
-
-if [ ! -z "$LDAP_INSECURE" ]; then
-    CMD = "$CMD --ldap-insecure"
-fi
-
-$CMD
```
Comment (johngmyers) [inline start.sh:30]: Unrelated change?

File: auth/token_issuer_test.go [hunk 5]
```diff
@@ -112,3 +115,76 @@ func TestCreateToken(t *testing.T) {
...
+
+	for i, c := range cases {
+		lti := LDAPTokenIssuer{
+			LDAPServer: "some-ldap-server",
+			TTL: c.TTL,
+		}
+
+		tok := lti.createToken(e)
+
+		time.Sleep(c.sleep)
```
Comment (johngmyers) [inline auth/token_issuer_test.go:183]: Normally for this sort of test one would mock out the clock through an interface. That would also allow the test to assert that the token doesn't expire too early.

File: start.sh (hunk 4 above)
Comment (rajatjindal) [reply start.sh:30]: yeah this is unrelated change. i should have opened a different pull request for this.

File: cmd/serve.go (hunk 2 above)
Comment (rajatjindal) [reply cmd/serve.go]: yep. Will revert this change

File: auth/webhook.go (hunk 3 above)
Comment (rajatjindal) [reply auth/webhook.go:46]: added test for this

File: auth/token_issuer_test.go (hunk 5 above)
Comment (rajatjindal) [reply auth/token_issuer_test.go:183]: golang by default don't provide a fake clock, and using a different interface will need more changes. 

in the interest of time, I will add an issue for it to be addressed later. does that sound ok? @johngmyers

Comment (johngmyers) [reply auth/token_issuer_test.go:183]: ok to address later. You'd just want to define an interface matching `time.Now()` and use time.Now() for the default constructed verifier.

**Review by johngmyers — APPROVED** [review]
